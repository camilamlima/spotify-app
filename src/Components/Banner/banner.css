.simplified{
  display: -moz-flex;
  display: -ms-flexbox;
  display: -ms-flex;
  display: -webkit-box;
  display: flex;
  -moz-flex-direction: column;
  -ms-flex-direction: column;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
          flex-direction: column;
  -moz-align-items: center;
  -ms-align-items: center;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -moz-justify-content: center;
  -ms-justify-content: center;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  min-height: 100%;
  padding-top: 0;
  padding-bottom: 0;
}

.center-text{
    position: relative;
    text-align: center;
    margin: 0 auto;
    z-index: 1;
    color: #fff;
}

.center-text h1 {
  font-size: 80px;
  font-weight: 900;
  letter-spacing: -0.04em;
  line-height: 1.15;
  margin-top: 0;
}
.center-text h4 {
    font-size: 18px;
    font-weight: 400;
}
.center-text .btn-green {
    color: #FFF;
    background-color: #1DB954;
    margin-top: 16px;
    border-radius: 500px;
    padding: 16px 48px 18px;
    -webkit-transition-property: background-color, border-color, color, -webkit-box-shadow, -webkit-filter;
    transition-property: background-color, border-color, color, -webkit-box-shadow, -webkit-filter;
    transition-property: background-color, border-color, color, box-shadow, filter;
    transition-property: background-color, border-color, color, box-shadow, filter, -webkit-box-shadow, -webkit-filter;
    -webkit-transition-duration: .3s;
    font-weight: 700;
    letter-spacing: 2px;
}

.background {
    background: linear-gradient(50deg, #cd5c5c 0, #ed8f42 100%);
    min-height: 80vh;
    position: relative;
    overflow: hidden;
    padding-top: 0;
    display: -moz-flex;
    display: -ms-flexbox;
    display: -ms-flex;
    display: -webkit-box;
    display: flex;
    -moz-flex-direction: row;
    -ms-flex-direction: row;
    -webkit-box-orient: horizontal;
    -webkit-box-direction: normal;
    flex-direction: row;
    
}

.hero-home-bg-cover {
    width: 1610px;
    height: 1523px;
    background: url(./hero-burst.svg) no-repeat;
    background-position: top -410px center;
    -webkit-transform: translate(-50%, 0%);
    -ms-transform: translate(-50%, 0%);
    transform: translate(-50%, 0%);

    position: absolute;
    left: 50%;
}
