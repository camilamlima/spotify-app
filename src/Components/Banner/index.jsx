import React, { Component } from 'react';

import './banner.css';

class Banner extends Component {
  render() {
    return (
       <div className="jumbotron jumbotron-devices background">
          <div className="hero-home-bg-cover"></div>  
          <div className="container simplified">
            <div className="row center-text">
              <h1>Música para todos.</h1>
              <h4>Milhões de músicas à sua escolha. E nem precisa de cartão de crédito.</h4>
              <a href="#" className="btn btn-green">
                OBTENHA O SPOTIFY FREE
              </a>
            </div>
          </div>
      </div>
      )
  }
}
export default Banner;