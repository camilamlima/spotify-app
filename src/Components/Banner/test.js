import React from 'react';
import { BrowserRouter } from "react-router-dom";
import renderer from 'react-test-renderer';

import Banner from '.';

test('Testar componente de Banner da aplicação', () => {

    const component = renderer.create(
      <BrowserRouter>
        <Banner />
      </BrowserRouter>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

});