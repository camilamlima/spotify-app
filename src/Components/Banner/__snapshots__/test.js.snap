// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testar componente de Banner da aplicação 1`] = `
<div
  className="jumbotron jumbotron-devices background"
>
  <div
    className="hero-home-bg-cover"
  />
  <div
    className="container simplified"
  >
    <div
      className="row center-text"
    >
      <h1>
        Música para todos.
      </h1>
      <h4>
        Milhões de músicas à sua escolha. E nem precisa de cartão de crédito.
      </h4>
      <a
        className="btn btn-green"
        href="#"
      >
        OBTENHA O SPOTIFY FREE
      </a>
    </div>
  </div>
</div>
`;
